### ISSUE TITLE
Backport docstring improvements to the C version of OrderedDict

### ISSUE THREAD

**Issue by rhettinger** [python/cpython#75526]
BPO | [31345](https://bugs.python.org/issue31345)
--- | :---
Nosy | @rhettinger, @ericsnowcurrently, @serhiy-storchaka, @Mariatta, @Cryvate
PRs | <li>python/cpython#3470</li><li>python/cpython#3471</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-09-11.01:12:22.255>
created_at = <Date 2017-09-04.23:06:40.934>
labels = ['easy', 'type-feature']
title = 'Backport docstring improvements to the C version of OrderedDict'
updated_at = <Date 2017-09-11.01:12:22.253>
user = 'https://github.com/rhettinger'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-11.01:12:22.253>
actor = 'Mariatta'
assignee = 'none'
closed = True
closed_date = <Date 2017-09-11.01:12:22.255>
closer = 'Mariatta'
components = []
creation = <Date 2017-09-04.23:06:40.934>
creator = 'rhettinger'
dependencies = []
files = []
hgrepos = []
issue_num = 31345
keywords = ['patch', 'easy']
message_count = 6.0
messages = ['301280', '301583', '301773', '301774', '301806', '301845']
nosy_count = 6.0
nosy_names = ['rhettinger', 'eric.snow', 'serhiy.storchaka', 'Mariatta', 'Sourav Singh', 'cryvate']
pr_nums = ['3470', '3471']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue31345'
versions = ['Python 3.6']
```

</p></details>

**Comment by rhettinger** [comment]
The help() for collections.OrderedDict.popitem is not showing the default value of last=True.

\----------- Prior versions are unhelpful \------------------
```pycon
~/npython $ python3.6
Python 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections import OrderedDict
>>> help(OrderedDict.popitem)
Help on method_descriptor:
```


```py
popitem(...)
    od.popitem() -> (k, v), return and remove a (key, value) pair.
    Pairs are returned in LIFO order if last is true or FIFO order if false.
```



\----------- Current version is helpful \--------------------
```pycon
$ ./python.exe
Python 3.7.0a0 (heads/master-dirty:b2d096bd2a, Sep  4 2017, 14:50:05)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections import OrderedDict
>>> help(OrderedDict.popitem)
Help on method_descriptor:
```


```py
popitem(self, /, last=True)
    Remove and return a (key, value) pair from the dictionary.
```


    Pairs are returned in LIFO order if last is true or FIFO order if false.

**Comment by 5761bba3-ba61-4f41-90eb-2f7a7e49dac3** [comment]
I am interested in working on the issue. Where do I need to make the fix?

**Comment by 7c3b9f7a-e616-458a-973e-3db6fe83c3bc** [comment]
I submitted a pull request for both 3.5 and 3.6 finding 2 locations using search and find. I also replayed Raymond's example (with \_collections as well) and got the documentation as expected on my custom builds of 3.5/3.6.

I submitted my CLA on 26th August but it has not been processed for some reason?

**Comment by Mariatta** [comment]
Thanks Henk-Jaap,
3.5 branch is now on security fix only, so only the backport to 3.6 will be accepted.

It seems like you have two bug tracker accounts? Did you register with two different email address?
It appears that the CLA is received for one of your account, but that's not the account you are using now.

**Comment by 7c3b9f7a-e616-458a-973e-3db6fe83c3bc** [comment]
Hi Mariatta,

Thanks for pointing that out about 3.5. Is that also why the build failed!? I found it hard to parse travis' output and was surprised it failed.

Thanks for letting me know---getting it fixed. I have added my github account to the other one, so that should fix it for this PR?

**Comment by Mariatta** [comment]
Thanks Henk-Jaap for the PR. It's been merged.
